api/api_server.go

```diff
@@ -29,13 +30,22 @@ func (s server) RequestStream(ctx context.Context, request *pb.StreamRequest) (*
 
 func (s server) RequestPremiere(ctx context.Context, request *pb.PremiereRequest) (*pb.Status, error) {
 	if request.WorkerID != cfg.WorkerID {
```

Comment:
In all these request handlers (`HandleStreamRequest`, `HandlePremiere`) you need to make sure, a pointer to the `StreamContext` is saved in `lectureHallStreams`.